lib/stack_master/stack_dependency.rb

```diff
@@ -11,22 +11,44 @@ def outdated_stacks
       @config.stacks.select do |stack|
         dependent_stack = Stack.find(stack.region, stack.stack_name)
         next unless dependent_stack
-        ParameterLoader.load(stack.parameter_files).any? do |_, value|
-          value['stack_output'] &&
-            value['stack_output'].gsub('_', '-') =~ %r(#{@stack_definition.stack_name}/) &&
-            outdated?(dependent_stack, value['stack_output'].split('/').last)
-        end
+        parameters = ParameterLoader.load(stack.parameter_files)
+        any_stack_output_outdated?(parameters, dependent_stack) || any_stack_outputs_outdated?(parameters, dependent_stack)
       end
     end
 
     private
 
+    def any_stack_output_outdated?(params, stack)
+      params.any? do |_, value|
+        value['stack_output'] &&
+          value['stack_output'].gsub('_', '-') =~ %r(#{@stack_definition.stack_name}/) &&
```

Comment:
These same cleanup calls are repeated in `any_stack_outputs_outdated?`.  Will it ever be needed elsewhere?  Shall we extract it?